# Reproduction of *Deep Mutational Scanning of an Oxygen-Independent Fluorescent Protein CreiLOV for Comprehensive Profiling of Mutational and Epistatic Effects*
Authors: Ryan Godin, Saman Ghazvini, William Amendola Bye, Jordyn Eovito, Vencke Gruening

## Paper Overview

This paper details the use of a combinatorial mutagenesis strategy to optimize the fluorescent properties of CreiLOV, a flavin mononucleotide-based fluorescent protein (FbFP.) FbFPs are advantageous compared to traditional fluorescent reporters such as GFP due to the lack of requirement for oxygen to produce fluorescence, making them applicable for use in anaerobic environments. One drawback is the relatively lower fluorescence generated by the protein when compared to other fluorescent proteins, along with a lowered quantum yield, leading the researchers to devise a high-throughput strategy to identify mutations that could alleviate some of these issues. Single mutants at each residue were created and evaluated for fluorescence, with a set of 15 residues and 20 specific mutations identified for their increased fitness and mutation tolerance. From this set, a combinatorial mutagenesis strategy was employed in order to assess epistatic effects, due to multiple mutations potentially interacting and changing the fluorescence of the protein. Employment of this strategy was successful in identifying the epistatic effects of multiple mutations on CreiLOV fluorescence and allowed for identification of specific mutants that may be suitable for serving as reporters in anaerobic contexts.

## Results Summary

### Figure 1B

In summary, this figure is depicting the pairwise relationships between replicate 1, replicate 2, and replicate 3. More specifically, the paper was interested
in examining the correlation between the fluorescence levels calculated by the simple mean between the three replicates. The first plot (rep1 vs rep2) indicates a correlation of 0.95,
meaning the data are highly positively correlated. The second plot (rep1 vs rep3) is also indicating a positive correlation given R = 0.957. 
Lastly, the third plot (rep2 vs rep3) is also indicating a positive correlation given R = 0.961, which is even greater than plots 1 and 2.

In comparison to the figure 1B results from the paper, we can see that our group got the same correlation coefficients, but different slopes for each of the regressions. This could be for various reasons, one being that the paper may have transformed their data in some sort of way that we did not (but this was not specified in the paper). Also, it seems as if they may have plotted a subsample of their data (the data we used had 164,000 entries) as their plots seem to have fewer points then ours. This is just a thought as the exact subsample that could have been used in the original analyses was not specified. Additionally, there was no git repository for this paper so we were not able to refer to source code or a markdown/readme file explaining their analyses and any data manipulation that was done for this figure.

### Figure 2A

Figure 2a shows a histogram of the estimated fluorescence intensities for all 2,185 single mutants of CreiLOV. The red bar marks the bin where the fluorescence value of the wild-type (WT) protein falls. The distribution shows that many single mutations had neutral or even positive effects on fluorescence.
 
For the replication of Figure 2a, the log_mean values from Supporting Dataset S1 were used as the fluorescence estimates for each variant. To highlight the bin where the wild-type (WT) genotype falls, the target value was set to 4.170360, which is the log_mean for the WT. No additional data processing or filtering was done.
 
Since the paper didn’t mention what bin size was used for the histogram, the bin size was estimated using the common rule of taking the square root of the total number of data points. This gave a histogram shape that looked similar to the published figure. After that, some formatting adjustments were made to get the figure to look as close as possible to the one in the paper.
 
The replicated figure closely matches the original in terms of overall shape and distribution. One key difference is on the y-axis: although the original figure labels it as "frequency," the axis only goes up to approximately 0.3, which does not align with raw counts. To address this, density was used when plotting the replicated figure, which resulted in the y-axis extending up to 1. The discrepancy between the two figures may be due to custom scaling or normalization of the density values in the original figure, which was not described in the paper. It is possible that the original figure plotted relative frequency (counts divided by the total number of observations) rather than absolute counts or density. Additionally, some plotting functions normalize histograms so that the area under the histogram equals 1 (probability density), while others normalize so that the sum of bar heights equals 1 (relative frequency). These two approaches produce slightly different bar heights. Several options were tested, including plotting density (after_stat(density)), frequency, and normalized count (after_stat(ncount)), but none of them replicated the exact y-axis scaling (0.3) seen in the original figure.

### Figure 2B

#### Heatmap

In order to replicate figure 2, mutations and fluorescence data were extracted from data set 1. The mutation data was parsed to extract the wildtype amino acid and positions, and the mutant amino acid and positions. Next, three letter amino acid codes were converted to one letter codes (for example “Arg”=”R”). The order of amino acids for the Y-axis was determined based on original figure. Heatmap was plotted using the ggplot function and WT fluorescence value was set as the median (white color), maximum fluorescence value was set to red and minimum was set to blue.

The heatmap appears to be overall consistent, with similar patterns emerging at positions 25, 45, and 70 for example.

#### Lower panel

In order to replicate the lower panel of figure 2b, first the mutation position numbers needed to be extracted from data set S1, which contains the single mutant data. A regular expression was used to extract the position number from the mutant name string, and this was saved as a new column in the data frame. As the mean fluorescence from each of the three replicates was already calculated as a column, the next step was to plot a boxplot of mean fluorescence values at each position in the protein. The position was used as a factor when plotting in order to group data for each position together, which resulted in a near-identical recreation of the original plot. 

### Figure 3A

In order to replicate figure 3a, more filtering had to be performed on both data sets. In the paper, 15 residues with 20 specific mutations were selected, and these were listed in data set S2 in rows 2-21. The specific mutation names were extracted from S2 and saved as a vector. Additionally, the 15 specific positions on the protein were manually entered into a vector to use for filtering. The mean value for the wildtype was extracted from S1 and saved as a value to be superimposed on the chart. In the paper, the graph shows the fluorescence plotted in black at each of the 15 positions, using all of the single-mutant data from S1, with the specific mutants listed in S2 plotted in red. 

One issue with reproducing the chart is the apparent lack of a data point for one of these specific mutations - Val107Met. Val107Met is listed in S2 as being one of the 20 selected mutations, but no single mutant value exists for Val107Met in S1. The red highlighted point at position 107, which should correspond to Val107Met is located above the mean wildtype fluoresence line (above 14,000 fluorescence units) but the data point in S2 for Val107Met is only at around 12,000 fluorescence units, indicating they did not use this point for the graph. It is unclear why this point is missing from the data set, as they do not explicitly list out the 20 specified mutants in the paper, rather the list had to be extracted from S2. From the single mutant data in S1, the 19 specified mutants (the 20 listed aside from Val107Met which doesn't exist in S1) were filtered out, so they would not be plotted twice. The selected mutants were filtered from S1 to a separate data frame to be plotted separately in red. The specific plot used wasn't specified, so a boxplot was used to graph the fluorescence at each point for the single-mutant data less the selected mutants. The bars shown in the paper are not error bars, as plotting the mean fluorescence +/- the standard deviation yields a much larger range for the bars, so it's not clear what the original authors had graphed. The mean wildtype fluorescence was plotted as a blue horizontal dashed line on top of the chart, and the selected mutant data points were plotted in red - note the lack of red point at position 107, corresponding to the lack of S1 data for Val107Met. 

Notably, there are data points in the recreated figure lying below the mean fluorescence line that do not exist in the original paper - it is unclear if these are outliers that the authors had removed prior to plotting, but there was no description of their data processing workflow that would indicate any specific type of data trimming. Overall, the reproduction looks very similar to the original figure, but without knowing what type of data trimming the authors did and what type of plot they used, it's difficult to get an exact carbon copy. 


### Figure 3B

In this figure, the fluorescence is plotted vs. the number of mutations. It is expected that as the number of mutations increase the fluorescent protein gets further from the wild-type and the likelihood of being fluorescent goes down. This is exactly what is observed with the authors original figure and our replication. They appear to match very closely.

### Figure 3D

This figure plots the level of epistasis, which is the difference between the observed fluorescence and expected fluorescence from summing the single mutations $e=\left(F_{\mathrm{com}}-F_{\mathrm{wt}}\right)-\Sigma\left(F_{\mathrm{sin}}-F_{\mathrm{wt}}\right)$, vs. the number of mutations. As expected the ability to predict the fluorescence gets harder the more mutations there are since they can interact with each other in unexpected ways. This is seen for the author's original figure and our figure. 

However, our figure does not completely line up with theirs. Specifically, there are more mutants with positive epistasis with the epsistasis appearing to be shifted up overall. We are unsure of the cause of this discrepancy.

### Figure 3E

This figure plots the strength of epistasis vs. the number of mutations. As expected the number of strong negative epistatic interactions get larger the more mutations there are. This is expected as the effect of mutations get harder to predict the more there are. It is also less likely the protein is fluorescent the farther it gets from the wild type. This is seen for the author's original figure and our figure which appear to match very closely.


